homeassistant/config_entries.py

```diff
@@ -25,6 +25,7 @@
 SOURCE_USER = "user"
 SOURCE_ZEROCONF = "zeroconf"
 SOURCE_IGNORE = "ignore"
+SOURCE_REDISCOVER = "rediscover"
```

Comment:
Could we add a comment here when this source is used (probably also add it to ignore)